File: golang/interpret/operation.go
package interpret

import "fmt"

func (o OpUnary_Operator) KeyName() string {
	switch o {
	default:
		panic(fmt.Sprintf("unexpected OperatorUnary %v", o))
	case OpUnary_LEN:
		return "len"
	case OpUnary_NOT:
		return "not"
	case OpUnary_FLAT:
		return "flat"
	case OpUnary_FLOOR:
		return "floor"
	case OpUnary_CEIL:
		return "ceil"
	case OpUnary_ABORT:
		return "abort"
	}
}

func (o OpBinary_Operator) KeyName() string {
	switch o {
	default:
		panic(fmt.Sprintf("unexpected OperatorBinary %v", o))
	case OpBinary_SUB:
		return "sub"
	case OpBinary_DIV:
		return "div"
	case OpBinary_MOD:
		return "mod"
	case OpBinary_EQ:
		return "eq"
	case OpBinary_NEQ:
		return "neq"
	case OpBinary_LT:
		return "lt"
	case OpBinary_LTE:
		return "lte"
	case OpBinary_GT:
		return "gt"
	case OpBinary_GTE:
		return "gte"
	}
}

func (o OpVariadic_Operator) KeyName() string {
	switch o {
	default:
		panic(fmt.Sprintf("unexpected OperatorVariadic %v", o))
	case OpVariadic_ADD:
		return "add"
	case OpVariadic_MUL:
		return "mul"
	case OpVariadic_AND:
		return "and"
	case OpVariadic_OR:
		return "or"
	case OpVariadic_CAT:
		return "cat"
	case OpVariadic_MIN:
		return "min"
	case OpVariadic_MAX:
		return "max"
	case OpVariadic_MERGE:
		return "merge"
	}
}

File: dart/lib/eval/operation.dart
import 'package:eyamluate/eval/eval.dart';

String opUnaryKeyName(OpUnary_Operator o) {
  return switch (o) {
    OpUnary_Operator.LEN => "len",
    OpUnary_Operator.NOT => "not",
    OpUnary_Operator.FLAT => "flat",
    OpUnary_Operator.FLOOR => "floor",
    OpUnary_Operator.CEIL => "ceil",
    OpUnary_Operator.ABORT => "abort",
    _ => throw "unexpected OperatorUnary $o",
  };
}

String opBinaryKeyName(OpBinary_Operator o) {
  return switch (o) {
    OpBinary_Operator.SUB => "sub",
    OpBinary_Operator.DIV => "div",
    OpBinary_Operator.EQ => "eq",
    OpBinary_Operator.NEQ => "neq",
    OpBinary_Operator.LT => "lt",
    OpBinary_Operator.LTE => "lte",
    OpBinary_Operator.GT => "gt",
    OpBinary_Operator.GTE => "gte",
    _ => throw "unexpected OperatorBinary $o",
  };
}

String opVariadicKeyName(OpVariadic_Operator o) {
  return switch (o) {
    OpVariadic_Operator.ADD => "add",
    OpVariadic_Operator.MUL => "mul",
    OpVariadic_Operator.AND => "and",
    OpVariadic_Operator.OR => "or",
    OpVariadic_Operator.CAT => "cat",
    OpVariadic_Operator.MIN => "min",
    OpVariadic_Operator.MAX => "max",
    OpVariadic_Operator.MERGE => "merge",
    _ => throw "unexpected OperatorVariadic $o",
  };
}
